cmd/satellite/main.go

```diff
@@ -1,63 +1,105 @@
 package main
 
 import (
+	"encoding/json"
 	"flag"
 	"log/slog"
 	"os"
 	"time"
 
+	"github.com/gpuctl/gpuctl/internal/config"
 	"github.com/gpuctl/gpuctl/internal/femto"
 	"github.com/gpuctl/gpuctl/internal/gpustats"
 	"github.com/gpuctl/gpuctl/internal/uplink"
 )
 
 func main() {
-	fakeGpus := flag.Bool("fakegpu", false, "Use fake GPU data")
-
-	// don't define flags below here
-	flag.Parse()
-	// don't access flags above here
+	flags := parseProgramFlags()
 
 	log := slog.Default()
 
-	log.Info("Starting satellite", "fakegpu", *fakeGpus)
+	log.Info("Starting satellite", "fakegpu", flags.fakeGPUs)
 
 	host, err := os.Hostname()
+
 	if err != nil {
 		log.Error("failed to get hostname", "err", err)
 		return
 	}
+
 	log.Info("got hostname", "hostname", host)
 
+	satellite_configuration, err := config.GetClientConfiguration("config.toml")
+
+	if err != nil {
+		log.Error("Failed to get satellite configuration from toml configuration file", "err", err)
```

Comment:
`Logger.Error` just logs, but doesn't exit. <https://pkg.go.dev/log/slog#Logger.Error>. If we fail to load, we should also exit the program